x-pack/legacy/plugins/ml/public/application/jobs/components/custom_url_editor/utils.js

```diff
@@ -7,6 +7,10 @@
 import { TIME_RANGE_TYPE, URL_TYPE } from './constants';
 
 import rison from 'rison-node';
+import url from 'url';
+
+import { npStart } from 'ui/new_platform';
+import { DASHBOARD_APP_LINK_GENERATOR } from '../../../../../../../../../src/plugins/dashboard_embeddable_container/public';
```

Comment:
Could it become a problem that we have to import this contract explicitly? 

Assuming we have AppFoo and AppBar which can link one to another. 
AppFoo imports LinkGenerators for AppBar
AppBar imports LinkGenerators for AppFoor

Will it work in NP if we make imports from the root of app plugins?

Reply:
I think this is okay.  An extremely similar example is mentioned here: https://github.com/elastic/kibana/issues/47065#issuecomment-564600529

And I believe it's not a problem because it's static code.  Does that sound accurate @joshdover ?